osu.Game/Online/Rooms/PlaylistItem.cs

```diff
@@ -96,6 +96,12 @@ public void MapObjects(RulesetStore rulesets)
         public bool ShouldSerializeID() => false;
         public bool ShouldSerializeapiBeatmap() => false;
 
-        public bool Equals(PlaylistItem other) => ID == other?.ID && BeatmapID == other.BeatmapID && RulesetID == other.RulesetID;
+        public bool Equals(PlaylistItem other)
+            => ID == other?.ID
+               && BeatmapID == other.BeatmapID
+               && RulesetID == other.RulesetID
+               && Expired == other.Expired
+               && allowedMods.SequenceEqual(other.allowedMods)
+               && requiredMods.SequenceEqual(other.requiredMods);
```

Comment:
I agree this is really bad. Hopefully we can improve this by splitting the api and non-api parts of the models.

I _think_ for the time being it shouldn't cause any problems since we're always calling `.Equals()` after mapping the mods across (e.g. `MultiplayerClient.createPlaylistItem()`).